src/main/java/Duke.java

```diff
@@ -1,10 +1,431 @@
+                } catch (IOException e) {
+                    System.out.println("Error when saving file: " + e.getMessage());
+                }
+                break;
+
+            case "deadline":
+                try {
+                    int taskEndIndex = inData.indexOf("/by");
+                    if (taskEndIndex == -1) { // no "/by"
+                        throw new InvalidDeadlineException();
+                    } else if (taskEndIndex <= DEADLINE_KEYWORD_LENGTH) { // no task title
+                        throw new InvalidTaskTitleException();
+                    } else if (taskEndIndex + 3 <= inData.length()) { // no task date
+                        throw new InvalidTaskDateException();
+                    }
+
+                    String deadlineTitle = inData.substring(DEADLINE_KEYWORD_LENGTH, taskEndIndex); 
+                    String deadlineDate = inData.substring(taskEndIndex + 3); //magic number, "/by"
+                    tasks.add(new Deadline(deadlineTitle, deadlineDate));
+
+                    System.out.println("Got it. I've added this deadline: \n" + "[D] [ ]"
+                            + deadlineTitle + " (at: " + deadlineDate + ")");
+                    System.out.println("Now you have " + tasks.size() +  " tasks in the list.");
+
+
+                    String newEntry = parseTaskToString(tasks);
+                    writeDisk(newEntry);
+
+
+                } catch (InvalidDeadlineException e){
+                    System.out.println("Error - '/by' must be included.");
+                } catch (InvalidTaskTitleException e) {
+                    System.out.println("Error - input a deadline description");
+                } catch (InvalidTaskDateException e) {
+                    System.out.println("Error - input a deadline date.");
+                } catch (IOException e) {
+                    System.out.println("Error when saving file: " + e.getMessage());
+                }
+                break;
+
+            case "delete":
+            try {
+                if ((inputTaskIndex) > tasks.size() || (inputTaskIndex) < 1) {
+                    throw new InvalidListIndexException();
+                }
+
+                String taskMark = getTaskMark(inLine);
+                String doneMark;
+                if (tasks.get(taskIndex).getTaskDone()){
+                    doneMark = "[X]";
+                } else{
+                    doneMark = "[ ]";
+                }
+                System.out.println("Noted. I have removed this task:");
+                System.out.println(taskMark + doneMark + " " +
+                        tasks.get(taskIndex).getTaskTitle());
+                System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+
+                tasks.remove(taskIndex);
+            } catch (InvalidListIndexException | NumberFormatException e) {
+                System.out.println("Error - List index given invalid. Please check again.");
+            }
+            break;
```

Comment:
You should consider indenting this block of code one level further, as stated in the coding standard.